client/driver/docker.go

```diff
@@ -1216,6 +1220,8 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 		Binds: binds,
 
 		VolumeDriver: driverConfig.VolumeDriver,
+
+		PidsLimit: driverConfig.PidsLimit,
```

Comment:
What happens when this is the default value for the type (0)? is it better not to set it?

Needs unit tests with the default and with setting it, and documentation. 

See PR https://github.com/hashicorp/nomad/pull/3512/files for an example of all the little things to change to add a new option to the docker driver